### PR TITLE
added sync api

### LINKED ISSUE #67: parseSync
Sometimes you need to parse the data on the main loop instead of in the background. I propose that we add `parseSync()` that does exactly that.

One use case is defining a `mocha` test for each endpoint in the blueprint.

```javascript
describe('API', function () {
  var ast = protagonist.parseSync(...).ast
  
  ast.....forEach(function (endpoint) {
    it('handels ' + endpoint.method, function (done) {
     ...
   })
  })
})
```

### PR DESCRIPTION
**Pull request by LinusU**
*Fixes #67*

This adds `parseSync` that parses on the main thread and returns the result.

It also fixes a previous existing test where it compared `result === result` instead of `result === expected`.

### CODE REVIEW

**Comment by pksunkara** [comment]
@LinusU Thanks for the PR. We are going to look into it. :balloon:

**Comment by LinusU** [comment]
Thank you :) I added usage information to the readme as well

**Comment by LinusU** [comment]
@pksunkara any news on this?

Since this repo has git submodules npm can't install my fork via `"protagonist": "LinusU/protagonist.git#c1b29e2"` which makes deploying a bitch currently...

I would really appreciate this getting merged :+1:

File: src/parse_async.cc [hunk 1]
```diff
@@ -67,38 +63,15 @@ NAN_METHOD(protagonist::Parse) {
...
-                // Unrecognized option
-                std::stringstream ss;
-                ss << "unrecognized option '" << *String::Utf8Value(key) << "', expected: ";
-                ss << "'" << RequireBlueprintNameOptionKey << "' or '" << ExportSourcemapOptionKey << "'";
-
-                NanThrowTypeError(ss.str().c_str());
-                NanReturnUndefined();
-            }
-        } 
+        optionsResult *res = ParseOptionsObject(Handle<Object>::Cast(args[1]));
```
Comment (pksunkara) [inline src/parse_async.cc]: Please rename this `res` variable as `optionsResult`

File: src/options_parser.cc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#include "snowcrash.h"
+
+using namespace v8;
+
+//static const std::string RenderDescriptionsOptionKey = "renderDescriptions";
+static const std::string RequireBlueprintNameOptionKey = "requireBlueprintName";
+static const std::string ExportSourcemapOptionKey = "exportSourcemap";
+
+optionsResult* ParseOptionsObject(Handle<Object> optionsObject) {
+    optionsResult *result = (optionsResult *) malloc(sizeof(optionsResult));
```
Comment (pksunkara) [inline src/options_parser.cc]: Please rename the `result` variable as `optionsResult`.

File: src/protagonist.h [hunk 3]
```diff
@@ -9,6 +9,13 @@
 #include "SectionParserData.h"
 #include "SourceAnnotation.h"
 
+struct optionsResult {
```
Comment (pksunkara) [inline src/protagonist.h]: Please rename the `optionsResult` struct as `OptionsResult`. And if possible, move this and the `ParseOptionsObject` function to be under `protagonist` namespace below.

File: src/parse_sync.cc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    // Get source data
+    String::Utf8Value sourceData(args[0]->ToString());
+
+    // Prepare options
+    snowcrash::BlueprintParserOptions options = 0;
+
+    if (args.Length() == 2) {
+        optionsResult *res = ParseOptionsObject(Handle<Object>::Cast(args[1]));
```
Comment (pksunkara) [inline src/parse_sync.cc]: Similarly, please rename `res` variable to `optionsResult`.

**Comment by pksunkara** [comment]
@LinusU Just a few minor variable renaming and this PR should be ready to merge. Thanks.

**Comment by LinusU** [comment]
Done and done :beers:

**Comment by pksunkara** [comment]
Thanks :smile:

**Comment by LinusU** [comment]
Perfect, if you could release this as `0.19.7` on npm I would be very happy :)

**Comment by JackuB** [comment]
Bump for releasing this on npm - I was scratching my head why `parseSync` isn't working for me. (It should be `0.20.0` though)

**Comment by zdne** [comment]
@pksunkara @kylef can you please release it?

**Pull request merged**